static/js/client.js

```diff
@@ -353,30 +357,47 @@ socket.on("fixq", function(message) {
                 var elt = $("#update_question_modal");
                 elt.find(".id-input").val(message.id);
                 M.Modal.getInstance(elt).open();
-            }
-            else if (ta_id) {
-                $(item).find(".remove-button").removeClass("hide");
+            } else if (ta_id) {
+                if (!ta_helping_id) {
+                    $(item).find(".remove-button").removeClass("hide");
+                    $(item).find(".help-button").removeClass("hide")
+                        .removeClass("waves-light btn blue")
+                        .addClass("waves btn-flat grey lighten-3 grey-text text-darken-2");
+                }
                 $(item).find(".helping-text").text("Student is updating question");
             }
         }
     });
 });
 
-socket.on("update-question", function(message) {
+socket.on("update-question-student", function(message) {
     if (disable_updates) return;
     checkAndUpdateSeq(message.seq);
+});
 
-    if (ta_id) { // A student's question is updated, reload
-        window.location.reload();
-        return;
-    }
+socket.on("update-question-ta", function(message) {
+    if (disable_updates) return;
+    checkAndUpdateSeq(message.seq);
```

Comment:
As mentioned above: you can combine these two, check if the user is a TA, and bail out early if not.